src/com/oltpbenchmark/BenchmarkState.java

```diff
@@ -1,132 +1,120 @@
- ******************************************************************************/
+/*
+ * Copyright 2020 by OLTPBenchmark Project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
 
 package com.oltpbenchmark;
 
+import com.oltpbenchmark.types.State;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
 import java.util.concurrent.CountDownLatch;
 import java.util.concurrent.atomic.AtomicInteger;
 
-import org.apache.log4j.Logger;
+public final class BenchmarkState {
 
-import com.oltpbenchmark.types.State;
+    private static final Logger LOG = LoggerFactory.getLogger(BenchmarkState.class);
 
-public final class BenchmarkState {
-    
-    private static final Logger LOG = Logger.getLogger(BenchmarkState.class);
-
-	private volatile State state = State.WARMUP;
-
-	// Assigned a value when starting the test. Used for offsets in the
-	// latency record.
-	private final long testStartNs;
-
-	public long getTestStartNs() {
-		return testStartNs;
-	}
-
-	private final CountDownLatch startBarrier;
-	private AtomicInteger notDoneCount;
-
-	// Protected by this
-
-	/**
-	 * 
-	 * @param numThreads
-	 *            number of threads involved in the test: including the
-	 *            master thread.
-	 * @param rateLimited
-	 * @param queueLimit
-	 */
-	public BenchmarkState(int numThreads) {
-		startBarrier = new CountDownLatch(numThreads);
-		notDoneCount = new AtomicInteger(numThreads);
-	
-		assert numThreads > 0;
-
-		testStartNs = System.nanoTime();
-	}
-
-	public State getState() {
-	    synchronized (this) {
-	        return state;
+    private final long testStartNs;
+    private final CountDownLatch startBarrier;
+    private final AtomicInteger notDoneCount;
+    private volatile State state = State.WARMUP;
+
+    /**
+     * @param numThreads number of threads involved in the test: including the
+     *                   master thread.
+     */
+    public BenchmarkState(int numThreads) {
+        startBarrier = new CountDownLatch(numThreads);
+        notDoneCount = new AtomicInteger(numThreads);
+
+
+        testStartNs = System.nanoTime();
+    }
+
+    // Protected by this
+
+    public long getTestStartNs() {
+        return testStartNs;
+    }
+
+    public State getState() {
+        synchronized (this) {
+            return state;
+        }
+    }
+
+    /**
+     * Wait for all threads to call this. Returns once all the threads have
+     * entered.
+     */
+    public void blockForStart() {
+
```

Comment:
TODO The asserts were removed? Granted Java asserts kind of suck since you need to remember to enable them

Reply:
Yeah I don't know how I feel about all the removed asserts. Revisit.

Reply:
Adding to description